Fails with node 18

Hey there!

When upgrading our application to node 18 it started throwing this error:

> /node_modules/.pnpm/moment-timezone@0.5.32/node_modules/moment-timezone/index.js:2:15-51 - Error: Module not found: Error: error:0308010C:digit
al envelope routines::unsupported

After some digging I believe the problem is in this plugin as it uses md4:

https://github.com/gilmoreorless/moment-timezone-data-webpack-plugin/blob/main/src/helpers.js#L149

Just trying to run this in node 18 throws the error:

```
console.log(require('crypto').createHash("md4"));
```

Happy to send a PR with the fix!

2 options I can see:

1. change the `md4` => `md5`, downsides to this is it will invalidate the cache when updating as the hash will change
2. conditionally change the hashing algorithm depending on the node version. More code but won't invalidate the cache on update

Let me know what you'd prefer!

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -21,7 +21,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node: [12, 14, 16]
+        node: [12, 14, 16, 18]
         # Only run one Node version on Windows
         include:
           - os: windows-latest

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,4 @@
 node_modules
 dist
 TODO.md
+.idea

--- a/src/helpers.js
+++ b/src/helpers.js
@@ -146,7 +146,7 @@ function cacheDir(cacheDirPath) {
 
 function cacheFile(tzdata, config, cacheDirPath) {
   const key = cacheKey(tzdata, config);
-  const filename = crypto.createHash('md4')
+  const filename = crypto.createHash('md5')
     .update(key)
     .digest('hex') + '.json';
   const filepath = path.join(cacheDir(cacheDirPath), filename);

--- a/test/utils.js
+++ b/test/utils.js
@@ -11,11 +11,12 @@ const rGeneratedFile = /[\\/]node_modules[\\/]\.cache[\\/]moment-timezone-data-w
 function buildWebpack(pluginOptions, testOptions = {}) {
   const compilerOptions = {
     mode: 'development',
-    devtool: 'hidden-source-map',
+    devtool: 'eval',
     entry: path.resolve(__dirname, 'fixtures', 'index.js'),
     output: {
       path: __dirname,
       filename: 'test-output.[filehash].js',
+      hashFunction: 'md5'
     },
     plugins: [
       new MomentTimezoneDataPlugin(pluginOptions),

